src/PhpDoc/TypeNodeResolver.php

```diff
@@ -525,6 +525,20 @@ private function resolveGenericTypeNode(GenericTypeNode $typeNode, NameScope $na
 			return new ErrorType();
 		} elseif ($mainTypeName === 'value-of') {
 			if (count($genericTypes) === 1) { // value-of<ValueType>
+				if ($genericTypes[0] instanceof ObjectType) {
+					$classReflection = $genericTypes[0]->getClassReflection();
+
+					if ($classReflection !== null && $classReflection->isBackedEnum()) {
+						$enumType = $classReflection->getBackedEnumType();
+
+						if ($enumType !== null) {
+							return $enumType;
+						}
+
+						return new ErrorType();
```

Comment:
It's a problem to return error type here. We might still have an object that is iterable so `value-of` should work for it too. Just remove this line.